src/tools.h

```diff
@@ -24,6 +24,36 @@
 #include <map>
 #include <string>
 #include <vector>
+#include <stdexcept>
+#include <sstream>
+
+/* Formatter for std::exception what() message:
+ * throw std::runtime_error(
+ *   Formatter() << "zimwriterfs: Unable to read" << filename << ": " << strerror(errno));
+ */
+class Formatter
```

Comment:
This class is nice. I may worth to move this in libzim's tools and use it everywhere we create a stringstream in our code.
This could even be merged with a `to_string` function.